pkg/util/common.go

```diff
@@ -977,13 +977,17 @@ func IsClusterRoleRuleExist(oldRules []rbacv1.PolicyRule, newRule rbacv1.PolicyR
 
 // GetRouteClient attempts to get a Route Client. It returns nil if it
 // fails due to an error or due to being on kubernetes (doesn't support routes)
-func GetRouteClient(restConfig *rest.Config) *routeclient.RouteV1Client {
+func GetRouteClient(restConfig *rest.Config) (*routeclient.RouteV1Client, error) {
 	routeClient, err := routeclient.NewForConfig(restConfig)
 	if err != nil {
-		log.Debugf("unable to get route client")
-		return nil
+		return nil, fmt.Errorf("unable to get route client")
+	} else {
```

Comment:
Golint indent: if block ends with a return statement, so drop this else and outdent its block. [More info](https://golang.org/wiki/CodeReviewComments#indent-error-flow). <!-- golint -->